### PR TITLE
Add database index for product.name Forums_Topic.Subject

### PR DESCRIPTION
**Pull request by curllog**
Hi this closes [https://github.com/nopSolutions/nopCommerce/issues/6940](https://github.com/nopSolutions/nopCommerce/issues/6940)

### CODE REVIEW

**Comment by skoshelev** [comment]
Hi @curllog. These changes are not sufficient because the Installation/Indexes.cs migration is only performed during a clean installation. It is also necessary to add new indexes to the file https://github.com/nopSolutions/nopCommerce/blob/develop/src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs and change the date in the migration to the current one

**Comment by curllog** [comment]
@skoshelev  I made the changes you requested hope it will be correct, Thanks

File: src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 namespace Nop.Data.Migrations.UpgradeTo470
 {
-    [NopSchemaMigration("2023-11-09 00:00:02", "AddIndexesMigration for 4.70.0")]
+    [NopSchemaMigration("2023-12-06 19:53:00", "Add Index For [Product].[Name]-[Forums_Topic].[Subject]")]
```
Comment (skoshelev) [inline src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs:14]: I think that the description of migration should not be changed, let it remain the original one.

File: src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs [hunk 2]
```diff
@@ -143,6 +143,20 @@ public override void Up()
                 IfDatabase(databaseType).Create.Index("IX_ProductReview_CustomerId").OnTable(nameof(ProductReview))
                     .OnColumn(nameof(ProductReview.CustomerId)).Ascending()
                     .WithOptions().NonClustered();
+
+            if(!Schema.Table(nameof(Product)).Index("IX_Product_Name").Exists())
+                IfDatabase(databaseType).Create.Index("IX_Product_Name")
```
Comment (skoshelev) [inline src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs:148]: There is no need to check the database type; these indexes must be added to all supported DBMSs

Comment (curllog) [reply src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs]: @skoshelev Done

**Comment by skoshelev** [comment]
@curllog Thank you for your help

**Comment by skoshelev** [comment]
I don’t think that we need to remove this, I believe that for those servers that do not support this concept, this parameter will be ignored, and for MsSql we need this index to be NonClustered

**Comment by curllog** [comment]
> I don’t think that we need to remove this, I believe that for those servers that do not support this concept, this parameter will be ignored, and for MsSql we need this index to be NonClustered

@skoshelev  I agree with you  and i think this is (and should be) handled in upstream library(fluentmigrator) since we are using database-agnostic namespace for migrations so do you recommend to open new issue and delete all IfDatabase(databaseType) from index migrations to 4.70? because it's fluentmigrator responsibility to check for supported migrations

**Comment by skoshelev** [comment]
no, we will not remove the IfDatabase check since indexes are created there that are not found only in MsSql, but which are present in two other DBMSs by default (but we do not know the names of these indexes, and in order not to index the same thing twice, we simply ignore all DBMS except MsSql)

**Pull request merged**